### DISCUSSION TITLE
What is the default scope for REST resource beans (using @Path annotation) ?

### DISCUSSION THREAD

**Discussion by omasseau** [quarkusio/quarkus#44387] [Q&A] [answered]
Let's say I have this : 

```
@Path
public class MyRestResource {
     //...
}
```

What will be the default scope ?

I cannot really find this information in the documentation, except it says that there will be only a single instance of the REST resource.
Does it mean that the default scope is `@ApplicationScoped` or `@Singleton`, as both are single instances ?

Is there any reason to explictitely use one of this scope explicitely ?
Which one is better ?

The documentation says that its better to use @ApplicationScoped : https://quarkus.io/guides/cdi#applicationscoped-and-singleton-look-very-similar-which-one-should-i-choose-for-my-quarkus-application

But I am not sure  if it applies to REST resources.

**Comment by mkouba** [comment, marked as answer]
> Does it mean that the default scope is @ApplicationScoped or @Singleton, as both are single instances ?

For `quarkus-resteasy` (RESTEasy Classic), it's controlled by the [`quarkus.resteasy.singleton-resources`](https://quarkus.io/guides/all-config#quarkus-resteasy-server_quarkus-resteasy-singleton-resources) config property and it should be `@Singleton` by default. For `quarkus-rest` (RESTEasy Reactive, default rest extension), it's `@Singleton` by default AFAIK. But maybe there are some exceptions? CC @geoand @FroMage.

> Is there any reason to explictitely use one of this scope explicitely ?
> Which one is better ?
> The documentation says that its better to use @ApplicationScoped : https://quarkus.io/guides/cdi#applicationscoped-and-singleton-look-very-similar-which-one-should-i-choose-for-my-quarkus-application
> But I am not sure if it applies to REST resources.

Indeed, this recommendation is intended for application developers. Extensions can decide to use `@Singleton`, for instance to achieve slightly better performance - because there is no indirection, no client proxy that delegates to the current instance from the context. Which makes sense for REST because resources are not regular beans, in the sense that they should not be injected anywhere.

In other words, it does not make sense to add `@ApplicationScoped` or `@Singleton` to a rest resource. The only meaningful alternative is to use `@RequestScoped` in order to create a new instance of the resource per HTTP request.

**Reply by geoand** [reply]
> For quarkus-rest (RESTEasy Reactive, default rest extension), it's @Singleton by default AFAIK.

This is absolutely correct
